app/src/test/java/com/codesoom/assignment/product/ui/ProductMockMvcControllerTest.java

```diff
@@ -0,0 +1,133 @@
+package com.codesoom.assignment.product.ui;
+
+import com.codesoom.assignment.product.application.ProductNotFoundException;
+import com.codesoom.assignment.product.application.ProductService;
+import com.codesoom.assignment.product.ui.dto.ProductResponseDto;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@DisplayName("ProductController 클래스")
+class ProductMockMvcControllerTest {
+    private static final Long PRODUCT_ID = 1L;
+    private static final String NAME = "snake";
+    private static final String MAKER = "cat toy";
+    private static final int PRICE = 5000;
+    private static final String IMAGE_URL = "https://http.cat/599";
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private ProductService productService;
+
+    @Autowired
+    protected ObjectMapper objectMapper;
+
+    @Nested
+    @DisplayName("GET /products는")
+    class Describe_getProducts {
+        @Nested
+        @DisplayName("등록된 상품이 있으면")
+        class Context_with_products {
+            List<ProductResponseDto> products;
+
+            @BeforeEach
+            void setUp() {
+                ProductResponseDto product = new ProductResponseDto(PRODUCT_ID, NAME, MAKER, PRICE, IMAGE_URL);
+                products = Arrays.asList(product);
```

Comment:
이렇게 한 개만 리스트로 만들 때에는 `Collections.singletonList`를 사용하는 것을 고려할 수 있습니다. 큰 차이는 없지만 토막 상식으로 알아둬도 괜찮을 거에요.

```suggestion
                products = Collections.singletonList(product);
```